pkg/gce-pd-csi-driver/controller.go

```diff
@@ -142,6 +152,44 @@ func isDiskReady(disk *gce.CloudDisk) (bool, error) {
 	return false, nil
 }
 
+// cloningLocationRequirements returns additional location requirements to be applied to the given create volume requests topology.
+// If the CreateVolumeRequest will use volume cloning, location requirements in compliance with the volume cloning limitations
+// will be returned: https://cloud.google.com/kubernetes-engine/docs/how-to/persistent-volumes/volume-cloning#limitations.
+func cloningLocationRequirements(req *csi.CreateVolumeRequest, cloneReplicationType string) (*locationRequirements, error) {
+	if !useVolumeCloning(req) {
+		return nil, nil
+	}
+	// If we are using volume cloning, this will be set.
+	volSrc := req.VolumeContentSource.GetVolume()
+	volSrcVolID := volSrc.GetVolumeId()
+
+	_, sourceVolKey, err := common.VolumeIDToKey(volSrcVolID)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "volume ID is invalid: %v", err.Error())
+	}
+
+	isZonalSrcVol := sourceVolKey.Type() == meta.Zonal
+	if isZonalSrcVol {
+		region, err := common.GetRegionFromZones([]string{sourceVolKey.Zone})
+		if err != nil {
+			return nil, fmt.Errorf("failed to get region from zones: %w", err)
```

Comment:
I instead changed the previous error return to just return fmt.Errorf rather than status.Errorf because the error returned from cloningLocationRequirements is already wrapped in InvalidArgument.  
```
locationTopReq, err = cloningLocationRequirements(req, params.ReplicationType)
if err != nil {
	return nil, status.Errorf(codes.InvalidArgument, "failed to get location requirements: %v", err.Error())
}
```